tests/test_views.py

```diff
@@ -915,6 +914,76 @@ def test_post_course_specific_consent_not_provided(
     @mock.patch('enterprise.views.configuration_helpers')
     @mock.patch('enterprise.views.render', side_effect=fake_render)
     @mock.patch('enterprise.views.CourseApiClient')
+    def test_post_course_specific_consent_not_provided_with_notification(
+            self,
+            course_api_client_mock,
+            *args  # pylint: disable=unused-argument
+    ):
+        # Verify that enterprise learner is redirected back to enterprise
+        # course enrollment page with consent decline warning message, if
+        # the learner declines data sharing consent.
+        self._login()
+        course_id = 'course-v1:edX+DemoX+Demo_Course'
+        course_name = 'edX Demo Course'
+        enterprise_customer = EnterpriseCustomerFactory(
+            name='Starfleet Academy',
+            enable_data_sharing_consent=True,
+            enforce_data_sharing_consent='at_enrollment',
+        )
+        enterprise_customer_user = EnterpriseCustomerUserFactory(
+            user_id=self.user.id,
+            enterprise_customer=enterprise_customer
+        )
+        enrollment = EnterpriseCourseEnrollment.objects.create(
+            enterprise_customer_user=enterprise_customer_user,
+            course_id=course_id
+        )
+        client = course_api_client_mock.return_value
+        client.get_course_details.return_value = {
+            'name': course_name
+        }
+        response = self.client.post(
+            self.url,
+            data={
+                'course_id': course_id,
+                'enterprise_customer_name': enterprise_customer.name,
+                'redirect_url': '/successful_enrollment',
+                'failure_url': '/failure_url?show_consent_decline_notification=true',
+            },
+        )
+        assert response.url.endswith('/failure_url')  # pylint: disable=no-member
+        assert response.status_code == 302
+        enrollment.refresh_from_db()
+        assert enrollment.consent_granted is False
+        expected_consent_decline_msg = '<span>We could not enroll you in <i>{course_name}</i>.</span> ' \
+                                       'If you have questions or concerns about sharing your data, please contact ' \
+                                       'your learning manager at {enterprise_customer_name}, or contact ' \
+                                       '<a href="{edx_enterprise_support_link}">edX support</a>.'.format(
+                                           course_name=course_name,
+                                           enterprise_customer_name=enterprise_customer.name,
+                                           edx_enterprise_support_link=EDX_ENTERPRISE_SUPPORT_URL,
+                                       )
+        # pylint: disable=protected-access, no-member
+        response_messages = messages.storage.cookie.CookieStorage(response)._decode(response.cookies['messages'].value)
```

Comment:
@zubair-arbi kindly take a look on that https://github.com/edx/ecommerce/blob/8fde125db17565c28cdf182fd35416c5c05b9dee/ecommerce/extensions/dashboard/tests.py#L8